js/ui/list/ui.list.edit.search.js

```diff
@@ -0,0 +1,170 @@
+            */
+            searchEnabled: false,
+
+            /**
+             * @name dxListOptions_searchEditorOptions
+             * @publicName searchEditorOptions
+             * @type TextBox options
+             * @default {}
+             */
+            searchEditorOptions: {}
+        });
+    },
+
+    _dataSourceOptions: function() {
+        return extend(this.callBase(), {
+            searchValue: this.option("searchValue"),
+            searchOperation: this.option("searchMode"),
+            searchExpr: this.option("searchExpr")
+        });
+    },
+
+    _render: function() {
+        this._renderSearch();
+        this.callBase();
+    },
+
+    _renderSearch: function() {
+        var editorOptions,
+            $element = this.element(),
+            searchEnabled = this.option("searchEnabled");
+
+        if(!searchEnabled) {
+            $element.removeClass(LIST_WITH_SEARCH_CLASS);
+            this._$searchEditorElement && this._$searchEditorElement.remove();
+            delete this._$searchEditorElement;
+            delete this._searchEditor;
+            return;
+        }
+
+        editorOptions = this._getSearchEditorOptions();
+
+        if(this._searchEditor) {
+            this._searchEditor.option(editorOptions);
+        } else {
+            $element.addClass(LIST_WITH_SEARCH_CLASS);
+            this._$searchEditorElement = $("<div>").addClass(LIST_SEARCH_CLASS).prependTo($element);
+            this._searchEditor = this._createComponent(this._$searchEditorElement, TextBox, editorOptions);
+        }
+    },
+
+    _getSearchEditorOptions: function() {
+        var that = this,
+            userEditorOptions = that.option("searchEditorOptions");
+
+        return extend({
+            mode: "search",
+            placeholder: messageLocalization.format("Search"),
+            tabIndex: that.option("tabIndex"),
+            value: that.option("searchValue"),
+            valueChangeEvent: "keyup",
+            onValueChanged: function(e) {
+                that.option("searchValue", e.value);
+            }
+        }, userEditorOptions);
+    },
+
+    _getAriaTarget: function() {
+        return this.element();
+    },
+
+    _focusTarget: function() {
+        if(this.option("searchEnabled")) {
+            return this._itemContainer();
+        }
+
+        return this.callBase();
+    },
+
+    _updateFocusState: function(e, isFocused) {
+        if(this.option("searchEnabled")) {
+            this._toggleFocusClass(isFocused, this.element());
+        }
+        this.callBase(e, isFocused);
+    },
+
+    _optionChanged: function(args) {
+        switch(args.name) {
+            case "searchEnabled":
+            case "searchEditorOptions":
+                this._invalidate();
+                break;
+            case "searchExpr":
+            case "searchMode":
+            case "searchValue":
+                if(!this._dataSource) {
+                    errors.log("W1009");
```

Comment:
dxList can work with "items" option and have no value in the "dataSource" option